src/Command/PelagosMigrateDatasetFilesCommand.php

```diff
@@ -0,0 +1,128 @@
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+class PelagosMigrateDatasetFilesCommand extends Command
+{
+    protected static $defaultName = 'pelagos:migrate-dataset-files';
+    
+    /**
+     * A Doctrine ORM EntityManager instance.
+     *
+     * @var EntityManagerInterface $entityManager
+     */
+    protected $entityManager;
+
+    /**
+     * Class constructor for dependency injection.
+     *
+     * @param EntityManagerInterface $entityManager A Doctrine EntityManager.
+     */
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+        // It is required to call parent constructor if
+        // using a constructon in a Symfony command.
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('This command will migrate the datastore to the new datastore.')
+            ->addOption('qf', null, InputOption::VALUE_NONE, 'Queue Dataset to Filer')
+            ->addOption('ifc', null, InputOption::VALUE_NONE, 'Do not check if dataset file exist on disk.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $io->title('Migrating Datasets Files');
+        
+        if ($input->getOption('qf')) {
+            $io->text('Also queueing the filer!');
+            $queueFiler = true;
+        } else {
+            $queueFiler = false;
+        }
+        
+        $datasetRepository = $this->entityManager->getRepository(Dataset::class);
+        
+        $datasets = $datasetRepository->findAll();
+        
+        foreach ($datasets as $dataset) {
+            $udi = $dataset->getUdi();
+
+            $status = $dataset->getDatasetStatus();
+            
+            $io->note(sprintf('UDI = %s.', $udi));
+            
+            if ($status === Dataset::DATASET_STATUS_IN_REVIEW) {
+                $datasetSubmission = $dataset->getLatestDatasetReview();
+                $this->setFile($datasetSubmission);
+                $datasetSubmission = $dataset->getDatasetSubmission();
+                $this->setFile($datasetSubmission);
+            } else {
+                $datasetSubmission = $dataset->getDatasetSubmission();
+                $this->setFile($datasetSubmission);
+            }
+            
+            $this->entityManager->persist($dataset);
+            
+        }
+
+        $this->entityManager->flush();
+        $io->success('Done!');
+
+        return 0;
+    }
+    
+    function setFile($datasetSubmission)
+    {
+        
+        if ($datasetSubmission instanceof DatasetSubmission) {
+            $udi = $datasetSubmission->getDataset()->getUdi();
+            
+            // TODO: Run querybuild to get filename and hash and size...
```

Comment:
won't the size and hash be the same anyway?

Reply:
No, we neutered the getHash and getName function to get them from the first file name in the files list. So it wasn't possible to get these, besides using a querybuilder.